### PR TITLE
[MTP] Implements JAX-native MTP drafter for Gemma4 

### PR DESCRIPTION
**Pull request by Lumosis**
# Description
This PR implements and registers the JAX-native **Gemma4 MTP (Multi-Token Prediction) draft/assistant model** (Drafter) in `tpu-inference`. It establishes the basic model architecture, parameter tree-based embedding sharing, and weights loading registry, and includes a comprehensive multi-host weight loading unit test suite.

> [!NOTE]
> **Scope Clarification**: This PR strictly covers the **MTP draft model architecture implementation, registry, and weight loading**. Full speculative decoding serving execution (speculative proposer orchestrators and runtime loops) for Gemma4 is *not* supported yet and will be integrated in subsequent PRs. 

#### 1. JAX-Native Gemma4 Speculative Draft Model (`tpu_inference/models/jax/gemma4_mtp.py`)
* Implemented the JAX-native speculative draft/assistant model (`Gemma4MTPForCausalLM` / `Gemma4MultiTokenPredictor` / `Gemma4MTPDecoderLayer`).
* Implemented **Q-only attention layers** (`Gemma4MTPAttention`) designed for cross-model KV cache sharing at runtime.
* Implemented **Centroid-Based Vocabulary Masking** (`Gemma4MTPMaskedEmbedder`) to sparsely calculate top-K centroid scores and dot-product vocabulary logits.

#### 2. Architecture Registration & Drafting Context (`tpu_inference/models/common/model_loader.py`)
* Registered `"Gemma4MTPModel"` under the standard JAX model architecture registry, mapping to JAX-native class `Gemma4MTPForCausalLM`.
* Adjusted `_get_nnx_model` to resolve the speculative draft configuration state dynamically during instantiations.

#### 3. Speculative Embedding Sharing (`tpu_inference/spec_decode/jax/eagle3.py`)
* Refactored speculative embedding sharing to dynamically query and bind JAX-native parameters directly using state parameter trees.
* Safely links target model token embeddings when draft embeddings are zeroed/missing, significantly reducing TPU memory footprint.

#### 4. Model Verification & Pipeline Parallel Tests (`tests/models/jax/test_gemma4_mtp.py`)
* Created a complete pipeline-parallel unit test suite specifically verifying the Gemma4 MTP draft model.
* Verifies model construction, weights loading, JAX device mesh array partitioning, and compiles/executes forward passes under pipeline parallel shapes (`pp_rank` and `pp_world_size` up to 4).
* Validates that weight loading correctly fails fast with a matching `ValueError` when serving with ordered embeddings enabled against standard (non-centroids) draft checkpoints.

---

### Test Plan & Results

Verified successfully on the Cloud TPU VM environment. 

#### Test Command:
```bash
python3 -m pytest -s -v tests/models/jax/test_gemma4_mtp.py
```

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Review by lk-chen — APPROVED** [review]
Would you mind sharing example serve command, and example output

**Review by gxd3 — APPROVED** [review]

**Comment by Lumosis** [comment]
> Would you mind sharing example serve command, and example output

This PR is about model implementation. The Gemma4 MTP support will be included in the subsequent PRs

**Review by wenxindongwork — APPROVED** [review]
